docker/client.py

```diff
@@ -747,6 +747,8 @@ def inspect_container(self, container):
 
     @check_resource
     def inspect_image(self, image):
+        if isinstance(image, dict):
+            image = image.get('Id')
```

Comment:
This isn't required to fix #602, but the change is make this method consistent with `inspect_container`.

@shin, incidentally, this also looks like a good candidate for inclusion in `@check_resource`. Similar checks are littered throughout `client.py` (but still missing from some methods). See my [raise-on-inspect-empty-string-602-with-id-resolution-in-check-resource](https://github.com/posita/docker-py/tree/raise-on-inspect-empty-string-602-with-id-resolution-in-check-resource) branch. (Tests pass.)